account_multicompany_easy_creation/wizards/multicompany_easy_creation.py

```diff
@@ -0,0 +1,342 @@
+            ('company_id', '=', self.new_company_id.id),
+            ('code', '=', account_template.code),
+        ], limit=1)
+
+    def set_global_properties(self):
+        IrProperty = self.env['ir.property'].sudo()
+        todo_list = [
+            ('property_account_receivable_id', 'res.partner',
+             'account.account',
+             self.match_account(self.account_receivable_id).id),
+            ('property_account_payable_id', 'res.partner',
+             'account.account',
+             self.match_account(self.account_payable_id).id),
+            ('property_account_expense_categ_id', 'product.category',
+             'account.account',
+             self.match_account(self.account_expense_categ_id).id),
+            ('property_account_income_categ_id', 'product.category',
+             'account.account',
+             self.match_account(self.account_income_categ_id).id),
+        ]
+        new_company = self.new_company_id
+        for record in todo_list:
+            if not record[3]:
+                continue
+            field = self.env['ir.model.fields'].search([
+                ('name', '=', record[0]),
+                ('model', '=', record[1]),
+                ('relation', '=', record[2])
+            ], limit=1)
+            vals = {
+                'name': record[0],
+                'company_id': new_company.id,
+                'fields_id': field.id,
+                'value': '{},{}'.format(record[2], record[3]),
+            }
+            properties = IrProperty.search([
+                ('name', '=', record[0]), ('company_id', '=', new_company.id)])
+            if properties:
+                properties.write(vals)
+            else:
+                IrProperty.create(vals)
+
+    def update_properties(self):
+        if self.smart_search_specific_account:
+            self.set_specific_properties('account.account', 'code')
+        if self.smart_search_fiscal_position:
+            self.set_specific_properties('account.fiscal.position', 'name')
+        if self.update_default_accounts:
+            self.set_global_properties()
+
+    def action_res_company_form(self):
+        action = self.env.ref('base.action_res_company_form').read()[0]
+        form = self.env.ref('base.view_company_form')
+        action['views'] = [(form.id, 'form')]
+        action['res_id'] = self.new_company_id.id
+        return action
+
+    def action_accept(self):
+        self.create_company()
+        self.update_taxes()
+        self.update_properties()
+        return self.action_res_company_form()
+
+
+class AccountMulticompanyBankWiz(models.TransientModel):
+    _inherit = 'res.partner.bank'
+    _name = 'account.multicompany.bank.wiz'
+    _order = 'id'
+
+    wizard_id = fields.Many2one(
+        comodel_name='account.multicompany.easy.creation.wiz',
+    )
+    partner_id = fields.Many2one(
+        default=lambda self: self.env.user.company_id.partner_id
```

Comment:
```suggestion
        required=False
```

Reply:
This is less confusing :wink:

Reply:
Yeah, that's what I was about to propose, but you need to add the company's partner in the dictionary preparation before calling the creation.